visualization/app/codeCharta/ui/codeMap/codeMap.arrow.service.ts

```diff
@@ -125,25 +127,20 @@ export class CodeMapArrowService
 
 		for (const edge of edges) {
 			const originNode = this.map.get(edge.fromNodeName)
-			// TODO: Maps should only have valid edges. If that's not the case, the
-			// internal decoration is likely faulty. Check if only test data is not
-			// correct or what the root cause of these checks actually is.
-			if (originNode === undefined) {
-				continue
-			}
```

Comment:
This code should be kept as it was before :-)
These continue conditions are already the safeguard :-)

Reply:
This will throw type errors for targetNode is undefined

Reply:
It seems we need to check for both again here

Reply:
This is handled a few lines below :-)

Reply:
sorry I ment originNode, that must have caused confusion
because in that case the next statement if (node.has(originNode.path)) would throw an error because originNode is undefined and therefore can not be asked for a path

Reply:
Therefore we need to check for both in the upper statement, at least with my local testing we get errors otherwise

Reply:
That part is never reached due to continue statement? What I meant is: this function should not have caused issues, only the function above.

Reply:
Yes I had another local change in there, 
I just reset all changes and pulled the branch again, so the result from changing 
`if (!originNode || !targetNode )`  to `if (originNode === undefined) ` will lead to type errors in targetNode  in line 137, so the check  `else if (node.has(targetNode.path))` because targetNode is not defined in that case

Reply:
The case where a originNode exists but has no path and we therefore add an arrow to the target node would be the case for nodes that are not the origin node (the ones receiving the arrow, visualized in light blue)
 if I understand the code correctly, which would fail in that case